### PR TITLE
Redis catch errors

### PR DESCRIPTION
**Pull request by EmadAlamoudi**
Also,
* MinMaxDistanceFunction -> MinMaxDistance in test_abc_smc_algorithm & test_samplerperf
* PercentileDistanceFunction -> PercentileDistance in test_abc_smc_algorithm & test_samplerperf

### CODE REVIEW

**Comment by EmadAlamoudi** [comment]
@yannikschaelte, can you please look at the Travis failure. I think it is something related to R & external module.

File: test/test_samplers.py [hunk 1]
```diff
@@ -225,3 +227,23 @@ def simulate_one():
...
+    def model(pars):
+        print(f"Model for {pars}")
+        if np.random.uniform() < 0.1:
+            raise ValueError("error")
+        return {'s0': pars['p0'] + 0.2 * np.random.uniform()}
+
+    def distance(s0, s1):
+        return abs(s0['s0'] - s1['s0'])
+    prior = Distribution(p0=RV("uniform", 0, 10))
+    sampler = RedisEvalParallelSampler(port=8775)
```
Comment (yannikschaelte) [inline test/test_samplers.py]: When you have the RedisEvalParallelSampler here, no workers are submitted. Instead, you should use the RedisEvalParallelSamplerServerStarter here.

File: test/test_samplers.py [hunk 2]
```diff
@@ -225,3 +227,23 @@ def simulate_one():
...
+    abc = ABCSMC(model, prior, distance, sampler=sampler)
+    db_file = "sqlite:///" + os.path.join(tempfile.gettempdir(), "test.db")
+    data = {'s0': 2.8}
+    abc.new(db_file, data)
+    with pytest.raises(Exception):
```
Comment (yannikschaelte) [inline test/test_samplers.py]: I would assume that in the new implementation nothing gets raised, i.e. you can just remove the ``with pytest.raises`` and just call run

File: test/test_samplers.py [hunk 3]
```diff
@@ -225,3 +227,23 @@ def simulate_one():
     sample = sampler.sample_until_n_accepted(10, simulate_one)
     assert 10 == len(sample.get_accepted_population())
     sampler.cleanup()
+
+
+def test_redis_catch_error():
+
+    def model(pars):
+        print(f"Model for {pars}")
```
Comment (yannikschaelte) [inline test/test_samplers.py]: No ``print`` needed here.

**Comment by yannikschaelte** [comment]
> @yannikschaelte, can you please look at the Travis failure. I think it is something related to R & external module.

Oh yes. There was an update of numpy (1.16.2 worked, 1.17.0 not) which cannot handle some data type any more, it seems. Again Rpy2 seems to fail. This is really tiresome.

File: test/test_samplers.py (hunk 2 above)
Comment (EmadAlamoudi) [reply test/test_samplers.py]: When removing `with pytest.raises`, the test method will fail then since the model create error

Comment (yannikschaelte) [reply test/test_samplers.py]: But the idea was that any errors are not propagated, but caught in the ``try-catch``, right?

File: pyabc/sampler/redis_eps/cli.py [hunk 4]
```diff
@@ -120,7 +120,11 @@ def work_on_population(redis: StrictRedis,
         # make batch_size attempts
         for n_batched in range(batch_size):
             # simulate
-            new_sim = simulate_one()
+            try:
+                new_sim = simulate_one()
+            except Exception as e:
+                logger.warning(f"Redis worker number:{n_worker} fail."
```
Comment (yannikschaelte) [inline pyabc/sampler/redis_eps/cli.py]:
```suggestion
                logger.warning(f"Redis worker number {n_worker} failed."
```

**Pull request merged**